mobile/src/main/java/org/openhab/habdroid/background/AlarmChangedReceiver.java

```diff
@@ -0,0 +1,72 @@
+package org.openhab.habdroid.background;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.util.Log;
+import androidx.work.Constraints;
+import androidx.work.NetworkType;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+
+import org.openhab.habdroid.R;
+
+import static android.content.Context.NOTIFICATION_SERVICE;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_ID_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND_ERROR;
+import static org.openhab.habdroid.background.BackgroundUtils.WORKER_TAG_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.background.BackgroundUtils.createNotificationChannels;
+import static org.openhab.habdroid.background.BackgroundUtils.makeBackgroundNotification;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ENABLED;
+
+public class AlarmChangedReceiver extends BroadcastReceiver {
+    private static final String TAG = AlarmChangedReceiver.class.getSimpleName();
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        Log.d(TAG, "onReceive() with intent " + intent.getAction());
+        if (!intent.getAction().equals("android.app.action.NEXT_ALARM_CLOCK_CHANGED")
+                || Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP
+                || ! PreferenceManager.getDefaultSharedPreferences(context)
+                .getBoolean(PREFERENCE_ALARM_CLOCK_ENABLED, true)) {
+            Log.d(TAG, "Don't send data");
+            return;
+        }
+
+        startAlarmChangedWorker(context);
```

Comment:
I'd suggest inverting the logic here, and use a constant for the action:
```
if (AlarmManager.ACTION_NEXT_ALARM_CLOCK_CHANGED.equals(intent.getAction())
        && Build.VERSION_SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
    SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
    if (prefs.getBoolean(PREFERENCE_ALARM_CLOCK_ENABLED, true))) {
        startAlarmChangedWorker(context);
    }
}
```

Reply:
IMO `false` is the better default for getBoolean, but otherwise I'll use it.